runtime/interpreter/value.go

```diff
@@ -16938,23 +16992,18 @@ func (v *DictionaryValue) Transfer(
 	remove bool,
 	storable atree.Storable,
 ) Value {
-	baseUse, elementOverhead, dataUse, metaDataUse := common.NewDictionaryMemoryUsages(
```

Comment:
Only meter if necessary. Is `AtreeMapPreAllocatedElementsMemoryUsage` enough though?

Reply:
@fxamacker I see. I tried to bring it back in [187ffc6](https://github.com/onflow/cadence/pull/2509/commits/187ffc6b45cb1c9f80dbeaf25758b0e16a163aac). Could you please have another look and see if that is correct?

Reply:
The update looks good! :+1: